lib/recurly/pricing/subscription/index.js

```diff
@@ -213,40 +201,42 @@ export default class SubscriptionPricing extends Pricing {
   /**
    * Updates coupon
    *
+   * `set.coupon` is emitted when a valid coupon is set
+   * `unset.coupon` is emitted when an existing valid coupon is removed
+   * `error.coupon` is emitted when the requested coupon is invalid
+   *
    * @param {String} newCoupon coupon code
    * @param {Object} newCoupon coupon object
    * @param {Function} [done] callback
    * @public
    */
   coupon (newCoupon, done) {
-    const unsetCoupon = () => {
-      debug('unset.coupon');
-      delete this.items.coupon;
-      this.emit('unset.coupon');
-    };
+    if (~this.couponCodes.indexOf(newCoupon)) {
```

Comment:
Same question as above ^

Reply:
common way of detecting if an element is in an Array